client/src/daemon.rs

```
@@ -112,8 +112,9 @@ pub fn run_wallet_daemon(force_testing_mode: bool) -> Result<()> {
         // Try load wallet. If no wallet make new.
         let mut wallet = match wallet::wallet::Wallet::load(wallet_data_loc, client_shim.clone()) {
             Ok(wallet) => wallet,
-            Err(_) => {
-                println!("No wallet file found. Creating new...");
+            Err(e) => {
```

If the wallet load error is that there is no file, the new wallet file should be created. However, for all other errors you should return the error and not create the new wallet - the wallet may be corrupted but recoverable, or the error could be some i/o error.